var/spack/repos/builtin/packages/python/package.py

```diff
@@ -432,11 +432,23 @@ def libs(self):
         library   = self.get_config_var('LIBRARY')
         ldlibrary = self.get_config_var('LDLIBRARY')
 
+        # MacOS/X Frameworks throw a spanner in the soup
+        libdir_prefix = os.path.dirname(libdir)
+        ldlibrary_dir = os.path.dirname(ldlibrary)
+        ldlibrary_name = os.path.basename(ldlibrary)
+        framework_library = ""
+        if len(libdir_prefix) and len(ldlibrary_dir) and len(ldlibrary_name) \
+           and libdir_prefix[-len(ldlibrary_dir):] == ldlibrary_dir \
+           and os.path.exists(os.path.join(libdir_prefix, ldlibrary_name)):
+            framework_ldlibrary = os.path.join(libdir_prefix, ldlibrary_name)
+
         # Prefer shared libraries
         if os.path.exists(os.path.join(libdir, ldlibrary)):
             return LibraryList(os.path.join(libdir, ldlibrary))
         elif os.path.exists(os.path.join(libdir, library)):
             return LibraryList(os.path.join(libdir, library))
+        elif os.path.exists(framework_ldlibrary):
+            return LibraryList(framework_ldlibrary)
```

Comment:
This would have to go above the static library checks if we still want to prefer shared libraries.